addons/payment/models/account_invoice.py

```diff
@@ -92,3 +92,11 @@ def payment_action_capture(self):
 
     def payment_action_void(self):
         self.authorized_transaction_ids.s2s_void_transaction()
+
+class AccountJournal(models.Model):
+    _inherit = 'account.journal'
+
+    @api.constrains('type')
+    def check_type(self):
+        if self.env['payment.acquirer'].search_count([('journal_id', 'in', self.filtered(lambda x : x.type not in ['bank', 'cash']).ids)]):
+            raise ValidationError(_("An acquirer is using this journal. Only bank and cash types are allowed."))
```

Comment:
```suggestion
    @api.constrains('type')
    def _check_journal_type_change(self):
        for journal in self:
            if journal.type not in ('bank', 'cash') and self.env['payment.acquirer'].search([('journal_id', 'in', self.ids)]):
                raise ValidationError(_("An acquirer is using this journal. Only bank and cash types are allowed."))
```

This way way, it will avoid unnecessary search if those are bank/cash.
If you prefer, you can filter the self from the for loop.